app/src/main/java/io/homeassistant/companion/android/background/LocationBroadcastReceiver.kt

```diff
@@ -126,25 +128,35 @@ class LocationBroadcastReceiver : BroadcastReceiver() {
         }
     }
 
-    private fun handleLocationUpdate(context: Context, intent: Intent) {
+    private suspend fun handleLocationUpdate(context: Context, intent: Intent) {
         Log.d(TAG, "Received location update.")
         LocationResult.extractResult(intent)?.lastLocation?.let {
-            sendLocationUpdate(it, context)
+
+            if (it.accuracy > integrationUseCase.getMinimumAccuracy()) {
+                Log.w(TAG, "GPS accuracy doesn't meet accuracy cutoff for location: $it")
+            } else {
+                sendLocationUpdate(it, context)
+            }
         }
     }
 
-    private fun handleGeoUpdate(context: Context, intent: Intent) {
+    private suspend fun handleGeoUpdate(context: Context, intent: Intent) {
         Log.d(TAG, "Received geofence update.")
         val geofencingEvent = GeofencingEvent.fromIntent(intent)
         if (geofencingEvent.hasError()) {
             Log.e(TAG, "Error getting geofence broadcast status code: ${geofencingEvent.errorCode}")
             return
         }
 
-        sendLocationUpdate(geofencingEvent.triggeringLocation, context)
+        if (geofencingEvent.triggeringLocation.accuracy > integrationUseCase.getMinimumAccuracy()) {
```

Comment:
I wonder if we should start with just having a hardcoded min accuracy and just launch with just this logic. That should already solve the issue that was reported.

Reply:
If we want to do that I can close this PR and start a new one that has a hardcoded value for minimum accuracy, it'll be easier than reverting the UI, Data, and Domain changes.  I do think that people are going to want different values based on the issues and some vocal people on Discord.
Just let me know if that is what you would like.

Reply:
Let's start by not customizing it. There will always be a minority vocal people that will shout that they want to be able to configure it, and in the end we have an app that only a tiny fraction can use because it's too complicated.

Reply:
Are you sure that option is overcomplicating the app? It's a simple value in meters with a description everybody is able to understand.
What would that default value be that accounts for all devices, location providers etc?

So far, only a minority of people asked not to be able to customize it.

Reply:
Do the best of both worlds. Keep a good default like we are at 50m and if people want to change it they can. Most users won't need to change an app setting if the default is good enough. People will get confused if they have to change the setting but not if it works out the box. Those who need more accurate location should be able to change it as needed.

Reply:
I would just keep it at 200. We literally had 1 or 2 reports that it was not working properly with geofencing and that's it? Seems a bit fast to jump to conclusions for an app with 30k installs.